instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/DynamoDbRequest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import static io.opentelemetry.instrumentation.awssdk.v2_2.FieldMapping.of;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.awssdk.core.SdkRequest;
+
+public enum DynamoDbRequest {
```

Comment:
While for an initial implementation it wouldn't be required, but ideally we generate this code from the YAML file itself, similar to how we generate the attribute constants

https://github.com/open-telemetry/opentelemetry-java/tree/main/buildscripts/semantic-convention

Do you think it could work?

For reference, aws xray sdk java currently uses a JSON file shared among the SDKs to define the mapping, but the OTel yaml should be able to serve the same purpose I believe.

https://github.com/aws/aws-xray-sdk-java/tree/master/aws-xray-recorder-sdk-aws-sdk-v2/src/main/resources/com/amazonaws/xray/interceptors

Reply:
Ah - it does mean reverse parsing "natural language" like "The value of the ... request parameter". I meant to be consistent for parsability, but it might not be, and it's definitely fragile. Hopefully we can find a structured way to model those rules though in that yaml.

Reply:
Makes perfect sense, manual converstion from spec into Java code just does not sound right. I'll open a follow-up issue for that.

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2291